tests/test_main.py

```diff
@@ -32,6 +32,7 @@ def workspace(self, auth_mock):
             workspace_mock.auth = auth_mock
             workspace_mock.id = constants.WORKSPACE_ID
             yield
+        main.workspace = main._Workspace()  # pylint: disable=protected-access
```

Comment:
The fixture `workspace` causes an interaction with other tests outside the TestClass. This means that if I run tests individually, they are fine. However, if I run tests altogether, there is an interaction due to the workspace singleton, and causes the `test_should_fail_if_not_authenticated` session tests to fail when run together with other tests.

For this reason, I needed to reset the fixture to the initial condition to make sure it did not affect other tests. This is not super elegant, but it solved the problem. Another solution I tested was to set the scope of the workspace fixture to "function" but I think the state of the singleton is not reset.

Reply:
It's unrelated, the reason is because main is authenticated at this point, therefore you need to patch it workspace down below in the class that fails similar how it is done here.